### PR TITLE
feat: Created SelectOption

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Added SelectOption

## **Manual testing steps**

1. Change pointer to storybook
2. Navigate to SelectOption

## **Screenshots/Recordings**
https://github.com/MetaMask/metamask-mobile/assets/14355083/dd0c9166-1f59-4e64-8c01-f533ad01165f

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Select/SelectOption/README.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# SelectOption
+
+SelectOption is a selectable option used in the SelectMenu and is part of the Select component.
+
+## Props
+
+This component extends [SelectValueProps](../SelectValue/SelectValue.types.ts) and [ListItemSelectProps](../../List/ListItemSelect/ListItemSelect.types.ts) from the [SelectValue](../SelectValue/SelectValue.tsx) and [ListItemSelect](../../List/ListItemSelect/ListItemSelect.tsx) components.
```
Comment (georgewrmarshall) [inline app/component-library/components/Select/SelectOption/README.md:7]: Checked all links work ✅

File: app/component-library/components/Select/SelectOption/SelectOption.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { default as SelectOptionComponent } from './SelectOption';
+import { SAMPLE_SELECTOPTION_PROPS } from './SelectOption.constants';
+
+const SelectOptionStoryMeta = {
+  title: 'Component Library / Select',
+  component: SelectOptionComponent,
+  argTypes: {
+    label: {
+      control: { type: 'text' },
+      defaultValue: SAMPLE_SELECTOPTION_PROPS.label,
```
Comment (georgewrmarshall) [inline app/component-library/components/Select/SelectOption/SelectOption.stories.tsx:15]: Did you manage to check if `defaultValue` is going to be deprecated in later versions for storybook in react native?

File: app/component-library/components/Select/SelectOption/SelectOption.test.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    expect(queryByRole('checkbox')).toBeNull();
+  });
+
+  it('should render the selected view if isSelected is true', () => {
+    const { queryByRole } = render(
+      <SelectOption {...SAMPLE_SELECTOPTION_PROPS} isSelected />,
+    );
+    expect(queryByRole('checkbox')).not.toBeNull();
+  });
+});
```
Comment (georgewrmarshall) [inline app/component-library/components/Select/SelectOption/SelectOption.test.tsx:28]: To improve the testing coverage for this component you could add tests for  `isDisabled `, `children`, `gap`, `verticalAlignment` props?

File: app/component-library/components/Select/SelectOption/SelectOption.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import ListItemSelect from '../../List/ListItemSelect/ListItemSelect';
+import SelectValue from '../SelectValue/SelectValue';
+
+// Internal dependencies.
+import styleSheet from './SelectOption.styles';
+import { SelectOptionProps } from './SelectOption.types';
+
+const SelectOption: React.FC<SelectOptionProps> = ({
+  style,
+  children,
```
Comment (georgewrmarshall) [inline app/component-library/components/Select/SelectOption/SelectOption.tsx:17]: `children` doesn't seem to be used in the component anywhere?

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left some minor non-blocking comments.
- Checked code 
- Checked documentation links, spelling and gramma
- Did not check running in simulator

Make sure you fill out the PR description properly you are missing:
- Manual testing steps
- Link to the related issue
- Checkboxes
PR descriptions greatly help with review and have massive pay offs in future when looking back on work and decisions made in the code base

File: app/component-library/components/Select/SelectOption/SelectOption.test.tsx (hunk 3 above)
Comment (brianacnguyen) [reply app/component-library/components/Select/SelectOption/SelectOption.test.tsx:28]: those are not the main prop of `SelectOption` but part of `ListItem` so it wont be included in this test

File: app/component-library/components/Select/SelectOption/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
+# SelectOption
+
+SelectOption is a selectable option used in the SelectMenu and is part of the Select component.
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/Select/SelectOption/README.md:3]: `SelectMenu` will most likely be the name of the component being rendered in the `BottomSheet`

File: app/component-library/components/Select/SelectOption/README.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
+# SelectOption
+
+SelectOption is a selectable option used in the SelectMenu and is part of the Select component
```
Comment (georgewrmarshall) [inline app/component-library/components/Select/SelectOption/README.md:3]: nit: formatting. Is `SelectMenu` a component or is it the "`Select` menu"
```suggestion
`SelectOption` is a selectable option used in the SelectMenu and is part of the Select component
```

**Review by georgewrmarshall — DISMISSED** [review]
Looks like you have some failing tests but code LGMT. Remember to fill out the PR description so we can understand what this PR contains in future

**Review by georgewrmarshall — COMMENTED** [review]
Code looks good but you haven't filled out the PR template

**Review by georgewrmarshall — DISMISSED** [review]
Need to fill out PR template otherwise LGTM

**Review by garrettbear — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**